onesignal-public.php

```diff
@@ -104,122 +52,118 @@ public static function onesignal_header()
+        if (array_key_exists('subdomain', $onesignal_wp_settings) && $onesignal_wp_settings['subdomain'] !== '') {
+            echo "oneSignal_options['subdomainName'] = \"".esc_html($onesignal_wp_settings['subdomain'])."\";\n";
         } else {
-            echo "oneSignal_options['path'] = \"".$current_plugin_url."sdk_files/\";\n";
+            echo "oneSignal_options['path'] = \"".esc_html($current_plugin_url)."sdk_files/\";\n";
         }
 
-        if (@$onesignal_wp_settings['safari_web_id']) {
-            echo "oneSignal_options['safari_web_id'] = \"".OneSignalUtils::html_safe($onesignal_wp_settings['safari_web_id'])."\";\n";
+        if (array_key_exists('safari_web_id', $onesignal_wp_settings) && $onesignal_wp_settings['safari_web_id']) {
+            echo "oneSignal_options['safari_web_id'] = \"".esc_html($onesignal_wp_settings['safari_web_id'])."\";\n";
         }
 
-        if ($onesignal_wp_settings['persist_notifications'] == 'platform-default') {
+        if (array_key_exists('persist_notifications', $onesignal_wp_settings) && $onesignal_wp_settings['persist_notifications'] === 'platform-default') {
             echo "oneSignal_options['persistNotification'] = false;\n";
-        } elseif ($onesignal_wp_settings['persist_notifications'] == 'yes-all') {
+        } elseif (array_key_exists('persist_notifications', $onesignal_wp_settings) && $onesignal_wp_settings['persist_notifications'] === 'yes-all') {
             echo "oneSignal_options['persistNotification'] = true;\n";
         }
 
         echo "oneSignal_options['promptOptions'] = { };\n";
-        if (array_key_exists('prompt_customize_enable', $onesignal_wp_settings) && $onesignal_wp_settings['prompt_customize_enable'] == '1') {
-            if ($onesignal_wp_settings['prompt_action_message'] != '') {
-                echo "oneSignal_options['promptOptions']['actionMessage'] = '".OneSignalUtils::html_safe($onesignal_wp_settings['prompt_action_message'])."';\n";
+        if (array_key_exists('prompt_customize_enable', $onesignal_wp_settings) && $onesignal_wp_settings['prompt_customize_enable'] === '1') {
+            if (array_key_exists('prompt_action_message', $onesignal_wp_settings) && $onesignal_wp_settings['prompt_action_message'] !== '') {
+                echo "oneSignal_options['promptOptions']['actionMessage'] = '".esc_html($onesignal_wp_settings['prompt_action_message'])."';\n";
             }
-            if ($onesignal_wp_settings['prompt_example_notification_title_desktop'] != '') {
-                echo "oneSignal_options['promptOptions']['exampleNotificationTitleDesktop'] = '".OneSignalUtils::html_safe($onesignal_wp_settings['prompt_example_notification_title_desktop'])."';\n";
+            if (array_key_exists('prompt_example_notification_title_desktop', $onesignal_wp_settings) && $onesignal_wp_settings['prompt_example_notification_title_desktop'] !== '') {
```

Comment:
Can we make helper for the blank string and existence check? Checking the key twice is making the if statements long and harder to read and maintain.